### PR TITLE
[Lens Migration] Migrate visualizations to lens for ActiveMQ, Spring Boot and WebSphere Application Server

### PR DESCRIPTION
**Pull request by rajvi-elastic**
- Enhancement

## What does this PR do?

- Migrate visualizations to lens for following integrations:
  - ActiveMQ
  - Spring Boot
  - WebSphere Application Server

Below are the statistics for **ActiveMQ** Lens migration:
<html>
<body>
<google-sheets-html-origin>

ActiveMQ Dashboards | Before Migration |   | After Migration |  
-- | -- | -- | -- | --
  | Lens | Visualization | Lens | Visualization
[Logs ActiveMQ] Log | 0 | 2 | 2 | 0
[Logs ActiveMQ] Audit | 0 | 2 | 1 | 1
[Metrics ActiveMQ] Broker | 0 | 6 | 6 | 0
[Metrics ActiveMQ] Topics | 0 | 8 | 8 | 0
[Metrics ActiveMQ] Queues | 0 | 8 | 8 | 0
  | 0 | 26 | 25 | 1

</body>
</html>

**Note:** One visualization is TagCloud that can not be migrated to Lens.

Below are the statistics for **Spring Boot** Lens migration:
<html>
<body>
<google-sheets-html-origin>

Spring Boot Dashboards | Before Migration |   | After Migration |  
-- | -- | -- | -- | --
  | Lens | Visualization | Lens | Visualization
[Spring Boot] Audit Events | 2 | 0 | 2 | 0
[Spring Boot] Overview Dashboard | 6 | 3 | 9 | 0
[Spring Boot] Access logs | 2 | 0 | 2 | 0
[Spring Boot] Memory and Threads | 5 | 0 | 5 | 0
  | 15 | 3 | 18 | 0

</body>
</html>

Below are the statistics for **WebSphere Application Server** Lens migration:
<html>
<body>
<google-sheets-html-origin>

WebSphere Application Server Dashboards | Before Migration |   | After Migration |  
-- | -- | -- | -- | --
  | Lens | Visualization | Lens | Visualization
[Metrics WebSphere Application Server] Session Manager | 4 | 1 | 4 | 0
[Metrics WebSphere Application Server] Servlet | 3 | 0 | 3 | 0
[Metrics WebSphere Application Server] ThreadPool | 6 | 1 | 6 | 0
[Metrics WebSphere Application Server] JDBC | 4 | 1 | 4 | 0
  | 17 | 3 | 17 | 0

</body>
</html>

**Note:** Three visualizations are migrated to new input controls.

## Checklist

- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [x] I have verified that panels are populated with data.
- [x] I have verified that panels are not distorted after being migrated to lens.
- [x] I have updated screenshots of dashboard.
- [x] I have verified that data count are matching and panel aggregations are same as before.

## Author's checklist

- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified an Kibana version is set to ^8.6.0.
- [x] Migrated panels should be removed from visualization folder. 

## Related issues

- Relates #5221

## Known issues
- Input Controls visualization is deprecated in Kibana and many of the dashboards contain these visualizations. Also, support for Input Controls has been stopped because of the new Input Controls. Hence, to replicate dashboard-level filters old Input Controls are replaced with new ones.
- TagCloud visualizations are not supported in the lens. [[GitHub issue](https://github.com/elastic/kibana/issues/95542)]
- Manually handle timestamp field in TSVB panels otherwise that panel will not be migrated using automation script.[[GitHub issue](https://github.com/elastic/kibana/issues/151109)]
- TSVB visualization doesn’t show negative values in UI. But the lens shows both negative and positive values. [[GitHub issue](https://github.com/elastic/elastic-charts/issues/135)]
- TSVB visualization doesn’t show values for the small intervals. But the lens provides data even for the small intervals. [[GitHub issue](https://github.com/elastic/kibana/issues/87704)] 
- Split chart functionality is not supported in Lens. [[GitHub issue](https://github.com/elastic/kibana/issues/82208)]
- While migrating the visualizations from TSVB to the lens, the colors are getting default ones instead of the configured ones.
- In the lens migrated metric visualizations, values are displayed in the bottom right corner. [[GitHub issue](https://github.com/elastic/kibana/pull/124124#discussion_r804476235)]
- Show toolbar option is not provided explicitly in Lens. Instead Download in CSV option is provided for the panels to download data. [[GitHub issue](https://github.com/elastic/kibana/pull/83430)]
- Panel titles are not displayed in lens metric visualization for Kibana `8.6.x` version. [[GitHub issue](https://github.com/elastic/kibana/issues/148563)]

### CODE REVIEW

File: packages/activemq/manifest.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ categories:
   - message_queue
   - infrastructure
 conditions:
-  kibana.version: ^8.2.0
```
Comment (lalit-satapathy) [inline packages/activemq/manifest.yml:17]: Is the Lens Migration going to force a Kibana version upgrade of the packages. Why is this being done? 

That may not be acceptable, if the packages are in the older releases.

**Comment by lalit-satapathy** [comment]
Regarding: packages/activemq/img/metricbeat-activemq-queues.png dont see the colour combinations of dashboard _before vs. after._ Is it related to this comment. What can we do to fix it?
>While migrating the visualizations from TSVB to the lens, the colors are getting default ones instead of the configured ones.

**Comment by rajvi-elastic** [comment]
> 

Yes @lalit-satapathy. 

- In TSVB below options are available in color palette:
cp-tsvb-1
cp-tsvb-2

- In Lens below only options are available in color palette:
cp-1
cp-2

In gradient color palette, user can customize the color ranges. But the gradient color palette is not available in lens. Hence, the exact color palette can not be replicated in lens. However, if colors are customized based on particular use case (for example, red color for failures and green color for success) then we are not relying on default colors and configuring the color ranges manually.

**Comment by stratoula** [comment]
Thanx @drewdaemon! One more question from my side. Can you give more details on 
> While migrating the visualizations from TSVB to the lens, the colors are getting default ones instead of the configured ones.

Do you mean with breakdown by Filters or another case? If yes, which?

**Comment by rajvi-crest** [comment]
> > TagCloud visualizations are not supported in the lens. [https://github.com/[elastic/kibana/issues/95542](https://github.com/elastic/kibana/issues/95542)]
> 
> Yes, we [suggest horizontal bar chart](https://github.com/elastic/integrations/pull/4915#issuecomment-1414368282) as a current alternative.
> 
**Make sense! We will migrate the TagCloud visualizations to horizontal bar chart.**
> > Manually handle timestamp field in TSVB panels otherwise that panel will not be migrated using automation script.[https://github.com/[elastic/kibana/issues/151109](https://github.com/elastic/kibana/issues/151109)]
> 
> We had trouble reproducing this issue, but got a response and may be able to look more into this. Is this a blocker or more of a nice-to-have?
> 
**No. This is not a blocker for us. But it would be good if this issue gets resolved. It seems like in TSVB, if some fields are not configured (eg., time_field, data-view etc.) then it will show the chart using default values (eg., @timestamp, logs-*). On the other hand, to migrate those visualizations in lens, these fields must be configured otherwise user won't see `Edit visualization in lens` option.** 
> > TSVB visualization doesn’t show negative values in UI. But the lens shows both negative and positive values. [https://github.com/[elastic/elastic-charts/issues/135](https://github.com/elastic/elastic-charts/issues/135)]
> 
> Looks like this will be fixed in 8.8.
> 
👍 
> > TSVB visualization doesn’t show values for the small intervals. But the lens provides data even for the small intervals. [https://github.com/[elastic/kibana/issues/87704](https://github.com/elastic/kibana/issues/87704)]
> 
> Are you sure the linked issue is related?
> 
**The issue is related to the default interval. Lens and TSVB shows the different output for the same interval (when interval is small and configured to "auto") [[GitHub Issue](https://github.com/elastic/kibana/issues/17843)]**
> > Split chart functionality is not supported in Lens. [https://github.com/[elastic/kibana/issues/82208](https://github.com/elastic/kibana/issues/82208)]
> 
> If you mean splitting a chart into multiple smaller charts by a breakdown dimension, this is the issue to track: [elastic/kibana#72063](https://github.com/elastic/kibana/issues/72063)
> 
> 
👍 
> > While migrating the visualizations from TSVB to the lens, the colors are getting default ones instead of the configured ones.
> 
> We'd welcome an issue if you haven't already filed one!
> 
**I have created one [GitHub](https://github.com/elastic/kibana/issues/152699) issue for this.**
> > In the lens migrated metric visualizations, values are displayed in the bottom right corner. [https://github.com/[elastic/kibana/pull/124124](https://github.com/elastic/kibana/pull/124124)#discussion_r804476235]
> 
> This according to the design of the new metric visualization. Metric always gets shown in bottom right to promote alignment between panels. See [this discussion](https://discuss.elastic.co/t/kibana-new-metric-visualization-too-small-and-doesnt-show-full-value/325077) for more details.
> 
👍 
> > Show toolbar option is not provided explicitly in Lens. Instead Download in CSV option is provided for the panels to download data. [https://github.com/[elastic/kibana/pull/83430](https://github.com/elastic/kibana/pull/83430)]
> 
> Is this a blocker/problem?
> 
**No. This is not a blocker issue. Because user is still able to export data using `Download as CSV` option.**
> > Panel titles are not displayed in lens metric visualization for Kibana 8.6.x version. [https://github.com/[elastic/kibana/issues/148563](https://github.com/elastic/kibana/issues/148563)]
> 
> True, this was by design. But they got added back in 8.7.
> 
👍

**Comment by rajvi-elastic** [comment]
We still need to work on this PR. Hence putting this in the draft as of now.

**Comment by rajvi-elastic** [comment]
Raised individual PRs for integrations and achieve minimum Kibana version using manual migration.

**Pull request closed without merging**